### ISSUE TITLE
Linter looking for vendor packages in `vendor/src` instead of `vendor`?

### ISSUE THREAD

**Issue by jpap** [Microsoft/vscode-go#1337]
I am getting lint errors on vendor packages, shown below.  This used to work on older versions with the same setup.  The project compiles on the command line without problems given the same `GOPATH` configured in VSC and on the command line.
```
cannot find package "github.com/spf13/afero" in any of:
    /usr/local/Cellar/go/1.9.2/libexec/src/github.com/spf13/afero (from $GOROOT)
    /Users/jpap/PROJECT/.gopath~/src/github.com/spf13/afero (from $GOPATH)
    /Users/jpap/PROJECT/vendor/src/github.com/spf13/afero
```

For brevity, assume `$PROJ=/Users/jpap/PROJECT`.

A workaround is to manually add a symlink in `$PROJ/vendor/src -> $PROJ/vendor`.

I am using [glide](https://github.com/Masterminds/glide) which places vendor packages in `$PROJ/vendor` not `$PROJ/vendor/src`.  Also, when glide refreshes the vendor tree, it removes the symlink workaround above.

My gopath is,
```
GOPATH=$PROJ/.gopath~:$PROJ/vendor
```
and where my project tree is as follows:
```
$PROJ
├ .gopath~
|  ├- src
|  |  └- mypackage -> $PROJ
... etc
```
This `GOPATH` is designed to allow for local paths like `$GOPATH/src/mypackage/file.go` but it should be unrelated to the `vendor/src` vs `vendor/` issue above?

Any ideas?

**Comment by ramya-rao-a** [comment]
Open the output pane, select "Go" from the drop down on the top right corner of the output pane.
You will see the commands used by the extension for running build, lint and vet.

Can you share the logs from this output pane?
This will help in figuring out which of the build/lint/vet process is giving the error.

Also run the `Go: Current GOPATH` command. It will show the GOPATH being used by the build/vet/lint process

**Comment by jpap** [comment]
Thanks for your help Ramya.  I've found the the issue is with the `go build` command: my command-line build works because I am building from the working directory `$PROJ/.gopath~/src/mypackage` whereas VSC is building from `$PROJ` because that's the `${workspaceRoot}`.

I cannot open `$PROJ/.gopath~/src/mypackage` directly into VSC because it is a symlink: VSC will instead open the `$PROJ` folder.

It used to work, but I guess a (recent?) change in behavior of golang broke this.  I can see now it doesn't directly relate to the Go VSC extension.

One solution would be an extension setting to select the working directory for the project that is different to `${workspaceRoot}`; is that currently possible?

**Comment by ramya-rao-a** [comment]
Just to ensure that this is not a regression in the Go extension, can you revert to a previous version of the extension and see if everything used to work fine?

You can download the vsix for the previous version from https://github.com/Microsoft/vscode-go/releases/tag/0.6.66

Then run `code --install-extension Go-0.6.66.vsix` and reload VS Code

**Comment by jpap** [comment]
It reproduces back on 0.6.59, so it does not look like a regression in the Go extension.  I have also tried various Go versions and it reproduces on all of them.  So it looks like either I never saw these lint errors before, or I ignored them.  :-) My apologies.

I recognize that my project structure is nonstandard; I can't think of a straightforward configuration setting that might help here: the `mypackage` symlink described above can be named anything, so specifying a configuration item as a path into it, and using it as a working directory, would then require passing to `go build` a partial-path relative to this for the source file opened in the editor and is not easily determined.

I'll close the issue until I can think of a better way to solve this.
